tests/Tools/EnvironmentUtilityTest.php

```diff
@@ -0,0 +1,66 @@
+<?php declare(strict_types=1);
+
+namespace PHPSu\Tests\Tools;
+
+use PHPSu\Exceptions\CommandExecutionException;
+use PHPSu\Tools\EnvironmentUtility;
+use PHPUnit\Framework\TestCase;
+
+class EnvironmentUtilityTest extends TestCase
+{
+    public function testCommandIsInstalled(): void
+    {
+        $this->assertEquals(true, (new EnvironmentUtility())->isCommandInstalled('echo'));
+    }
+
+    public function testCommandIsNotInstalled(): void
+    {
+        $this->assertEquals(false, (new EnvironmentUtility())->isCommandInstalled('reiguheruh'));
+    }
+
+    public function testGetInstalledRsyncVersionOrExpectExceptionIfNotInstalled(): void
+    {
+        if ((new EnvironmentUtility())->isRsyncInstalled()) {
+            $rsyncVersion = (new EnvironmentUtility())->getRsyncVersion();
+            $this->assertEquals(1, version_compare($rsyncVersion, '2.0.0'));
+        } else {
+            $this->expectException(CommandExecutionException::class);
+            (new EnvironmentUtility())->getRsyncVersion();
+        }
+    }
+
+    public function testGetInstalledSshVersionOrExpectExceptionIfNotInstalled(): void
+    {
+        if ((new EnvironmentUtility())->isSshInstalled()) {
+            $sshVersion = (new EnvironmentUtility())->getSshVersion();
+            $this->assertEquals(1, version_compare($sshVersion, '3.0.0'));
+        } else {
+            $this->expectException(CommandExecutionException::class);
+            (new EnvironmentUtility())->getSshVersion();
+        }
+    }
+
+    public function testGetInstalledMysqldumpVersionOrExpectExceptionIfNotInstalled(): void
+    {
+        if ((new EnvironmentUtility())->isMysqlDumpInstalled()) {
```

Comment:
mark test as risky, as it dose not test the functions in every case?

Reply:
it tests in every case: if installed, i am expecting a version result, if not installed i expect an exception